db/migrate/20240521100629_add_vault_credential_id_to_options_in_service_templates.rb

```diff
@@ -0,0 +1,44 @@
+class AddVaultCredentialIdToOptionsInServiceTemplates < ActiveRecord::Migration[6.1]
+  class ServiceTemplate < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.table_name = "service_templates"
+    self.inheritance_column = :_type_disabled
+    serialize :options
+  end
+
+  def up
+    ServiceTemplate.all.each do |record|
+      update_vault_credential_id(record, :provision)
+      update_vault_credential_id(record, :retirement)
+    end
+  end
+
+  def down
+    ServiceTemplate.all.each do |record|
+      clear_vault_credential_id(record, :provision)
+      clear_vault_credential_id(record, :retirement)
+    end
+  end
+
+  private
+
+  def update_vault_credential_id(record, action)
+    if record.options.present? && record.options[:config_info].present? && record.options[:config_info][action].present?
+      vault_credentials = record.options[:config_info][action][:vault_credentials]
+      if vault_credentials.present?
+        record.options[:config_info][action][:vault_credential_id] = vault_credentials
+        record.update!(:options => record.options)
+      end
+    end
```

Comment:
I think this can be more simply:

```suggestion
    vault_credentials = record.options&.dig(:config_info, action, :vault_credentials)
    return unless vault_credentials.present?
    
    record.options[:config_info][action][:vault_credential_id] = vault_credentials
    record.save!
```